src/match_regex.c

```diff
@@ -295,6 +371,24 @@ static int mr_match (const data_set_t __attribute__((unused)) *ds, /* {{{ */
 	if (mr_match_regexen (m->type_instance,
 				vl->type_instance) == FC_MATCH_NO_MATCH)
 		return (nomatch_value);
+	if (vl->meta != NULL)
+	{
+		for (llentry_t *e = llist_head(m->meta); e != NULL; e = e->next)
+		{
+			mr_regex_t *meta_re = (mr_regex_t *) e->value;
+			char *value;
+			int status = meta_data_get_string (vl->meta, e->key, &value);
+			if (status == 0)  /* key is present */
```

Comment:
Is this actually the correct behavior? If I set `MetaData "foo" "^bar"` and a metric does *not* have the "foo" meta data at all, I'd expect the that to be a no-match.